ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/JdbcLedgerDao.scala

```diff
@@ -141,6 +147,141 @@ private class JdbcLedgerDao(
     ()
   }
 
+  private val SQL_UPDATE_CURRENT_CONFIGURATION = SQL(
+    "update parameters set configuration={configuration}"
+  )
+  private val SQL_SELECT_CURRENT_CONFIGURATION = SQL("select configuration from parameters")
+
+  private val SQL_GET_CONFIGURATION_ENTRIES = SQL(
+    "select * from configuration_entries where ledger_offset>={startInclusive} and ledger_offset<{endExclusive} order by ledger_offset asc")
+
+  private def updateCurrentConfiguration(configBytes: Array[Byte])(
+      implicit conn: Connection): Unit = {
+    SQL_UPDATE_CURRENT_CONFIGURATION
+      .on("configuration" -> configBytes)
+      .execute()
+    ()
+  }
+
+  private def selectLedgerConfiguration(implicit conn: Connection) =
+    SQL_SELECT_CURRENT_CONFIGURATION
+      .as(byteArray("configuration").?.single)
+      .flatMap(Configuration.decode(_).toOption)
+
+  override def lookupLedgerConfiguration(): Future[Option[Configuration]] =
+    dbDispatcher.executeSql("lookup configuration")(implicit conn => selectLedgerConfiguration)
+
+  private val configurationEntryParser: RowParser[(Long, ConfigurationEntry)] =
+    (long("ledger_offset") ~
+      str("typ") ~
+      str("submission_id") ~
+      str("participant_id") ~
+      str("rejection_reason")(emptyStringToNullColumn).? ~
+      byteArray("configuration"))
+      .map(flatten)
+      .map {
+        case (offset, typ, submissionId, participantIdRaw, rejectionReason, configBytes) =>
+          val config = Configuration
```

Comment:
I like to do the decoding of the protobuf messages as part of processing the akka stream (compare `getLedgerEntries`), because then
1) the decoding only happens when demand for more messages is requested from the sink (i.e. the API consumer)
2) we don't do all the work upfront just to find out that the stream was broken.

Admittedly this is less of a problem for configuration than for transactions, but it'll ultimately be more sane to use the same "process".

For now, don't worry about it.

Reply:
agree with the reasoning. I'll leave it as is for now though to land this beast...